apps/touch/main.c

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2022, 2023 - Tillitis AB
+// SPDX-License-Identifier: GPL-2.0-only
+
+#include <tk1_mem.h>
+#include <types.h>
+
+// clang-format off
+static volatile uint32_t *led =   (volatile uint32_t *)TK1_MMIO_TK1_LED;
+static volatile uint32_t *touch = (volatile uint32_t *)TK1_MMIO_TOUCH_STATUS;
+
+#define LED_BLACK 0
+#define LED_RED   (1 << TK1_MMIO_TK1_LED_R_BIT)
+#define LED_GREEN (1 << TK1_MMIO_TK1_LED_G_BIT)
+#define LED_BLUE  (1 << TK1_MMIO_TK1_LED_B_BIT)
+// clang-format on
+
+void wait_touch_ledflash(int ledvalue, int loopcount)
+{
+	int led_on = 0;
+	// first a write, to ensure no stray touch?
+	*touch = 0;
+	for (;;) {
+		*led = led_on ? ledvalue : 0;
+		for (int i = 0; i < loopcount; i++) {
+			if (*touch & (1 << TK1_MMIO_TOUCH_STATUS_EVENT_BIT)) {
+				goto touched;
```

Comment:
goto? Ouch!

Reply:
This is just copy/paste from the signer https://github.com/tillitis/tillitis-key1-apps/blob/main/apps/signer/main.c#L26

I think use of goto is OK if it's used forward-only to break out of nested loops or so.

Reply:
Yeah, just a copy from signer. Didn't prioritize to rewrite it.